cocos/core/assets/texture-cube.ts

```diff
@@ -112,6 +155,64 @@ export class TextureCube extends SimpleTexture {
         }
     }
 
+    set mipmapAtlas (value) {
+        this._mipmapAtlas = value;
+        if (!this._mipmapAtlas) {
+            this.reset({
+                width: 0,
+                height: 0,
+                mipmapLevel: 0,
+            });
+            return;
+        }
+        const imageAtlasAsset: ImageAsset = this._mipmapAtlas.atlas.front;
+        if (!imageAtlasAsset.data) {
+            return;
+        }
+        const faceAtlas = this._mipmapAtlas.atlas;
+        const layout = this._mipmapAtlas.layout;
+        const mip0Layout = layout[0];
+
+        const ctx = Object.assign(document.createElement('canvas'), {
+            width: imageAtlasAsset.width,
+            height: imageAtlasAsset.height,
+        }).getContext('2d');
+
+        this.reset({
+            width: mip0Layout.width,
+            height: mip0Layout.height,
+            format: imageAtlasAsset.format,
+            mipmapLevel: layout.length,
+        });
+
+        for (let j = 0; j < layout.length; j++) {
+            const layoutInfo = layout[j];
+            _forEachFace(faceAtlas, (face, faceIndex) => {
+                ctx!.clearRect(0, 0, imageAtlasAsset.width, imageAtlasAsset.height);
+                const drawImg = face.data as HTMLImageElement;
+                ctx!.drawImage(drawImg, 0, 0);
```

Comment:
This is a platform related approach, you should modify SimpleTexture API to support upload a region of an ImageAsset directly.

```
// Functions need to be modified in simple-texture.ts
protected _assignImage (image: ImageAsset, region: BufferTextureCopy, level: number, arrayIndex?: number);
public uploadData (source: HTMLCanvasElement | HTMLImageElement | ArrayBufferView | ImageBitmap, region: BufferTextureCopy,, level = 0, arrayIndex = 0)

// Also need to modify _assignImage in texture-cube.ts to upload region for a specific face
this._assignImage(atlas, region, level, faceIndex);
```